bodhi/client/__init__.py

```diff
@@ -444,12 +447,26 @@ def download(url, **kwargs):
 
     Args:
         staging (bool): Whether to use the staging server or not.
+        allarch (bool): Whether to download packages for all architectures or only for current one.
+        arch (unicode): Requested architecture of packages to download.
         url (unicode): The URL of a Bodhi server to create the update on. Ignored if staging is
                        True.
         kwargs (dict): Other keyword arguments passed to us by click.
     """
     client = bindings.BodhiClient(base_url=url, staging=kwargs['staging'])
+    requested_arch = kwargs['arch']
+
+    if kwargs['allarch'] is True and requested_arch is None:
+        all_arch = True
+    if kwargs['allarch'] is True and requested_arch is not None:
+        all_arch = False
+        click.echo("WARNING: --allarch suppressed by --arch")
```

Comment:
I think this approach is fine if you prefer it, but here are a couple other ideas:

0. Make it an error to specify both flags.
1. Have only the ```--arch``` flag and have it accept an "all" option.

I think I'd probably go for the second of those two, but if you really like the current approach that's fine with me.